### PR TITLE
Feature file service inclusion

### PR DESCRIPTION
**Pull request by iAKM**
- File upload feature implemented
- Dev testing is done with the dev system
- Unit tests are failing at the moment. Need to change the existing unit tests
- Execution strategy is written in readme file

Additional ENV variables to be added:
UPLOAD_TOPIC (the topic to send to once the upload is completed)

### CODE REVIEW

File: src/controller/osw-controller.ts [hunk 1]
```diff
@@ -16,10 +46,13 @@ class GtfsOSWController implements IController {
         this.intializeRoutes();
     }
 
+    eventBusService = new EventBusService();
+
     public intializeRoutes() {
         this.router.get(this.path, this.getAllOsw);
         this.router.get(`${this.path}/:id`, this.getOswById);
-        this.router.post(this.path, this.createOsw);
+        this.router.post(this.path, validationMiddleware(OswVersions), this.createOsw);
```
Comment (susrisha) [inline src/controller/osw-controller.ts:54]: remove this line. Its a duplicate implementation

File: src/scripts/generate-db.sql [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+-- DROP TABLE IF EXISTS public.osw_versions;
+
+CREATE TABLE IF NOT EXISTS public.osw_versions
+(
+    id integer NOT NULL DEFAULT nextval('osw_versions_id_seq'::regclass),
+    tdei_record_id character varying COLLATE pg_catalog."default" NOT NULL,
+    confidence_level integer DEFAULT 0,
+    tdei_org_id character varying COLLATE pg_catalog."default" NOT NULL,
+    tdei_service_id character varying COLLATE pg_catalog."default" NOT NULL,
```
Comment (susrisha) [inline src/scripts/generate-db.sql:25]: remove this. `OSW` does not have `tdei_service_id`

**Review by susrisha — CHANGES_REQUESTED** [review]

File: src/controller/osw-controller.ts (hunk 1 above)
Comment (iAKM) [reply src/controller/osw-controller.ts]: fixed

File: src/scripts/generate-db.sql (hunk 2 above)
Comment (iAKM) [reply src/scripts/generate-db.sql]: fixed

**Review by iAKM — COMMENTED** [review]
fixed requested changes

File: src/controller/osw-controller.ts [hunk 3]
```diff
@@ -61,38 +93,56 @@ class GtfsOSWController implements IController {
...
-                            }
-                        });
-                }
-            });
         } catch (error) {
-            console.error('Error saving the osw version', error);
-            response.status(500).send('Error saving the osw version')
-            next(new HttpException(500, "Error saving the osw version"));
+            console.error('Error saving the osw file', error);
+            response.status(500).send('Error saving the osw file');
```
Comment (susrisha) [inline src/controller/osw-controller.ts:145]: Please check the code from other repo. It should respond with `next` if the `error` is of type `HTTPException`

File: src/service/storage-service.ts [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+     * @param body Readable stream of the body
+     * @param containerName Name of the container. defaults to gtfs-osw
+     */
+    async uploadFile(filePath:string, type:string = 'application/zip' ,body:NodeJS.ReadableStream, containerName:string = 'osw'){
+        const client = Core.getStorageClient();
+        const container = await client?.getContainer(containerName);
+        const file = container?.createFile(filePath, type);
+        const uploadedEntity = await file?.upload(body);
+        // need to get the full path here and return
+        // console.log('File path',uploadedEntity!.filePath);
```
Comment (susrisha) [inline src/service/storage-service.ts:42]: please remove this commented code

**Review by susrisha — CHANGES_REQUESTED** [review]

File: src/service/storage-service.ts (hunk 4 above)
Comment (iAKM) [reply src/service/storage-service.ts]: fixed

File: src/controller/osw-controller.ts (hunk 3 above)
Comment (iAKM) [reply src/controller/osw-controller.ts]: fixed

**Review by susrisha — APPROVED** [review]

**Pull request merged**